pgxn/neon_walredo/inmem_smgr.c

```diff
@@ -63,15 +67,26 @@ static void inmem_open(SMgrRelation reln);
 static void inmem_close(SMgrRelation reln, ForkNumber forknum);
 static void inmem_create(SMgrRelation reln, ForkNumber forknum, bool isRedo);
 static bool inmem_exists(SMgrRelation reln, ForkNumber forknum);
-static void inmem_unlink(RelFileNodeBackend rnode, ForkNumber forknum, bool isRedo);
-static void inmem_extend(SMgrRelation reln, ForkNumber forknum,
-						 BlockNumber blocknum, char *buffer, bool skipFsync);
+static void inmem_unlink(NRelFileInfoBackend rinfo, ForkNumber forknum, bool isRedo);
 static bool inmem_prefetch(SMgrRelation reln, ForkNumber forknum,
 						   BlockNumber blocknum);
+#if PG_MAJORVERSION_NUM < 16
+static void inmem_extend(SMgrRelation reln, ForkNumber forknum,
+						 BlockNumber blocknum, char *buffer, bool skipFsync);
 static void inmem_read(SMgrRelation reln, ForkNumber forknum, BlockNumber blocknum,
 					   char *buffer);
 static void inmem_write(SMgrRelation reln, ForkNumber forknum,
 						BlockNumber blocknum, char *buffer, bool skipFsync);
+#else
+static void inmem_extend(SMgrRelation reln, ForkNumber forknum,
+						 BlockNumber blocknum, const void *buffer, bool skipFsync);
+static void inmem_zeroextend(SMgrRelation reln, ForkNumber forknum,
```

Comment:
inmem_zeroextend is not implemented. I guess that's OK because none of the WAL redo functions use it? But a dummy implementation that does nothing or just throws an error would be in order.